core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1216,15 +1228,21 @@ class LogManager(logDirs: Seq[File],
             cleaner.updateCheckpoints(removedLog.parentDirFile, partitionToRemove = Option(topicPartition))
           }
         }
-        removedLog.renameDir(UnifiedLog.logDeleteDirName(topicPartition), false)
+        if (isStray) {
```

Comment:
Looks like we are only renaming the stray log dir and removing it from LogManager. Do we also want to add the logic on delayed stray log deletion? If we want to delete the stray logs immediately (more risky, and it might create conflicts with AK merge), I think we need to add it to the log deletion queue (by calling `addLogToBeDeleted(strayLog)`).

Right now the log dir will be renamed to "-stray" but it will not be deleted by the broker.

Reply:
For now, the desired behavior is to stop tracking the partition but not delete the files. Since migrations are are one-off and inherently risky, I didn't want to take any destructive actions like deleting the logs (immediately or delayed). The stray'd partitions are logged at the INFO level when they are detected, and at WARN on subsequent startups. 

This gives give operators the information needed to clean up stray partitions if desired. 

I filed https://issues.apache.org/jira/browse/KAFKA-15698 to track automatic clean up of the stray partitions.

Reply:
Got it. Makes sense.